oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/handler/ZipkinSpanHTTPHandler.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.skywalking.oap.server.receiver.zipkin.ZipkinReceiverConfig;
+import org.apache.skywalking.oap.server.receiver.zipkin.trace.SpanForward;
+import org.apache.skywalking.oap.server.telemetry.TelemetryModule;
+import org.apache.skywalking.oap.server.telemetry.api.CounterMetrics;
+import org.apache.skywalking.oap.server.telemetry.api.HistogramMetrics;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsCreator;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsTag;
+import zipkin2.Span;
+import zipkin2.codec.SpanBytesDecoder;
+
+import static java.util.Objects.nonNull;
+
+@Slf4j
+public class ZipkinSpanHTTPHandler {
+    private final ZipkinReceiverConfig config;
+    private final SourceReceiver sourceReceiver;
+    private final NamingControl namingControl;
+    private final HistogramMetrics histogram;
+    private final CounterMetrics errorCounter;
+
+    public ZipkinSpanHTTPHandler(ZipkinReceiverConfig config, ModuleManager manager) {
+        sourceReceiver = manager.find(CoreModule.NAME).provider().getService(SourceReceiver.class);
+        namingControl = manager.find(CoreModule.NAME).provider().getService(NamingControl.class);
+        this.config = config;
+        MetricsCreator metricsCreator = manager.find(TelemetryModule.NAME)
+                                               .provider()
+                                               .getService(MetricsCreator.class);
+        histogram = metricsCreator.createHistogramMetric(
+            "trace_in_latency", "The process latency of trace data",
+            new MetricsTag.Keys("protocol"), new MetricsTag.Values("zipkin")
+        );
+        errorCounter = metricsCreator.createCounter(
+            "trace_analysis_error_count", "The error number of trace analysis",
+            new MetricsTag.Keys("protocol"), new MetricsTag.Values("zipkin")
+        );
+    }
+
+    @Post("/api/v2/spans")
+    public HttpResponse collectV2Spans(ServiceRequestContext ctx, HttpRequest req) {
+        return doCollectSpans(SpanBytesDecoder.JSON_V2, ctx, req);
+    }
+
+    @Post("/api/v2/spans")
+    @ConsumesJson
+    public HttpResponse collectV2JsonSpans(ServiceRequestContext ctx, HttpRequest req) {
+        return doCollectSpans(SpanBytesDecoder.JSON_V2, ctx, req);
+    }
+
+    @Post("/api/v2/spans")
+    @ConsumesProtobuf
+    public HttpResponse collectV2ProtobufSpans(ServiceRequestContext ctx, HttpRequest req) {
+        return doCollectSpans(SpanBytesDecoder.PROTO3, ctx, req);
+    }
+
+    @Post("/api/v1/spans")
+    public HttpResponse collectV1Spans(ServiceRequestContext ctx, HttpRequest req) {
+        return doCollectSpans(SpanBytesDecoder.JSON_V1, ctx, req);
+    }
+
+    @Post("/api/v1/spans")
+    @ConsumesJson
+    public HttpResponse collectV1JsonSpans(ServiceRequestContext ctx, HttpRequest req) {
+        return doCollectSpans(SpanBytesDecoder.JSON_V1, ctx, req);
+    }
+
+    @Post("/api/v1/spans")
+    @ConsumesThrift
```

Comment:
Could you point out the doc to explain this? I think `x-thrift` is just a decoding process in Zipkin. Why does this relate to HTTP server?

Reply:
You can see there are several methods mapped to the same endpoint `@Post("/api/v1/spans")`, this annotation `@ConsumesThrift` means only those requests with http header `Content-Type: application/x-thrift` will be handled by this method, meaning only thrift stream will be decoded by this method, correctly.

Reply:
Here we are using the annotation `@Consumes**` to route the different protocols to responding decoder.